models/articles.models.js

```diff
@@ -0,0 +1,39 @@
+const db = require("../db/connection");
+const fs = require("fs/promises")
+
+exports.fetchArticleById = (article_id) => {
+   return fs.readFile('./endpoints.json', 'utf-8').then((fileContents) => {
+      const previousEndpoints = JSON.parse(fileContents)
```

Comment:
I like the idea with this to programmatically add in the endpoint with readFile & writeFile, however I'm going to say this is unnecessary for a couple of reasons:
- This code only needs to run once, after that it's overwriting the same info every time anyone makes a request to this endpoint
- It would need to be added to every other model for every other endpoint, totalling a lot of extra code

My advice would be to remove this logic and manually add the new endpoints into the endpoints files, think of it more as a readme that you are updating with new info for users.